commons/src/test/java/org/infinispan/commons/util/MutableIntSetTest.java

```diff
@@ -0,0 +1,479 @@
+      intSet.add(1);
+      intSet.add(4);
+
+      intSet2.add(1);
+      intSet2.add(4);
+      intSet2.add(7);
+
+      assertFalse(intSet.containsAll(intSet2));
+      assertTrue(intSet2.containsAll(intSet));
+   }
+
+   @Test
+   public void testContainsAll1() throws Exception {
+      intSet.add(1);
+      intSet.add(4);
+
+      Set<Integer> set = CollectionFactory.makeSet(1, 4, 7);
+
+      assertFalse(intSet.containsAll(set));
+      assertTrue(set.containsAll(intSet));
+   }
+
+   @Test
+   public void testAddAll() throws Exception {
+      intSet.add(1);
+      intSet.add(4);
+
+      IntSet intSet2 = new SmallIntSet();
+      intSet2.addAll(intSet);
+      assertEquals(2, intSet2.size());
+   }
+
+   @Test
+   public void testAddAll1() throws Exception {
+      IntSet rangeSet = new RangeSet(4);
+
+      intSet.addAll(rangeSet);
+      assertEquals(4, intSet.size());
+   }
+
+   @Test
+   public void testAddAll2() throws Exception {
+      intSet.addAll(CollectionFactory.makeSet(1, 4));
+      assertEquals(2, intSet.size());
+   }
+
+   @Test
+   public void testRemoveAll() throws Exception {
+      intSet.add(1);
+      intSet.add(4);
+      intSet.add(7);
+
+      assertEquals(3, intSet.size());
+
+      IntSet intSet2 = new SmallIntSet();
+      intSet2.add(4);
+      intSet2.add(5);
+      intSet2.add(7);
+
+      assertTrue(intSet.removeAll(intSet2));
+
+      assertEquals(1, intSet.size());
+      assertTrue(intSet.contains(1));
+   }
+
+   @Test
+   public void testRemoveAll1() throws Exception {
+      intSet.add(1);
+      intSet.add(4);
+      intSet.add(7);
+
+      assertEquals(3, intSet.size());
+
+      // (1 - 4)
+      IntSet rs = new RangeSet(5);
+
+      assertTrue(intSet.removeAll(rs));
+
+      assertEquals(1, intSet.size());
+      assertTrue(intSet.contains(7));
+   }
+
+   @Test
+   public void testRemoveAll2() throws Exception {
+      intSet.add(1);
+      intSet.add(4);
+      intSet.add(7);
+
+      assertEquals(3, intSet.size());
+
+      assertTrue(intSet.removeAll(CollectionFactory.makeSet(4, 5, 7)));
+
+      assertEquals(1, intSet.size());
+      assertTrue(intSet.contains(1));
+   }
+
+   @Test
+   public void testRetainAll() throws Exception {
+      intSet.add(1);
+      intSet.add(4);
+      intSet.add(7);
+
+      assertEquals(3, intSet.size());
+
+      IntSet intSet2 = new SmallIntSet();
+      intSet2.add(4);
+      intSet2.add(5);
+      intSet2.add(7);
+
+      intSet.retainAll(intSet2);
+
+      assertEquals(2, intSet.size());
+      assertTrue(intSet.contains(4));
+      assertTrue(intSet.contains(7));
+   }
+
+   @Test
+   public void testRetainAll1() throws Exception {
+      intSet.add(1);
+      intSet.add(4);
+      intSet.add(7);
+
+      assertEquals(3, intSet.size());
+
+      // (0 - 4)
+      IntSet rs = new RangeSet(5);
+
+      assertTrue(intSet.retainAll(rs));
+
+      assertEquals(2, intSet.size());
+      assertTrue(intSet.contains(1));
+      assertTrue(intSet.contains(4));
```

Comment:
I see you didn't change the assertions

Reply:
Hrmm, not sure what you wanted for this one. I changed a lot of the assertions :)

Reply:
In the comment above, `assertEquals(hashSetOf(1, 4), intSet)`